### PR TITLE
IA-338: add translations to QuestionMappingForm

### PR DESCRIPTION
**Pull request by quang-le**
## Changes

- Used react-intl to replace hard-coded text
- Added corresponding translations
- Added some missing dependencies in useEffect dep arrays
- Replaced `<button>` tags with MaterialUI `Button`

@mestachs Can you pls double check the french translations?

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/app/translations/fr.json [hunk 1]
```diff
@@ -349,11 +347,38 @@
...
+    "iaso.mappings.label.mapping": "Correspondance: {name}, {id} - {type}",
+    "iaso.mappings.label.noMapping": "pas de correspondance",
+    "iaso.mappings.label.options": "Options :",
+    "iaso.mappings.label.program": "Programme",
+    "iaso.mappings.label.programDataElement": "Data element du programme",
+    "iaso.mappings.label.proposedNewMapping": "Nouvelle correspondance proposée",
+    "iaso.mappings.label.relationshipType": "Type de relation",
+    "iaso.mappings.label.searchDataElement": "Cherecher un data element par nom, code ou id (et combinaisons)",
+    "iaso.mappings.label.searchTrackedEntity": "Rechercher l'attribut type de l'entité",
+    "iaso.mappings.label.searchTracker": "Rechecher le tracker data element par nom, code ou id",
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/app/translations/fr.json:370]: @mestachs Can you check this translation please?

File: hat/assets/js/apps/Iaso/domains/app/translations/en.json [hunk 2]
```diff
@@ -10,13 +9,13 @@
     "blsq.treeview.search.options.label.noOptions": "No result found",
     "blsq.treeview.search.results.label.display": "Display",
     "blsq.treeview.search.results.label.resultsLower": "result(s)",
-    "iaso.CDS": "Health centre",
-    "iaso.SSC": "Community Health Site",
     "iaso.algo.addRun": "Add algorithm run",
     "iaso.archived.title": "Archives",
     "iaso.area": "Health area",
+    "iaso.CDS": "Health centre",
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/domains/app/translations/en.json:15]: I think you editor sort the JSON keys while ignoring case and mine sort it with upper case first, we should probably decide a convention if we don't want to pass our time going back and forth

Comment (quang-le) [reply hat/assets/js/apps/Iaso/domains/app/translations/en.json:15]: Can you change your settings easily or is it a pain?

**Comment by olethanh** [comment]
doesn't seem so, I have looked but didn't find any settings for it.

**Pull request merged**